### PR TITLE
fix(amplify-cli): promise not resolving in lts/dubnium

### PR DESCRIPTION
**Pull request by roychoo**
promise not awaited, resulting aws-exports not generating.

*Issue #, if available:*
https://github.com/aws-amplify/amplify-cli/issues/186
https://github.com/aws-amplify/amplify-cli/issues/1051

*Description of changes:*
return promise from ```copyBatch``` to the top.

please do let me know how can i create a test for this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by roychoo** [comment]
Hi, sorry for the chaser. any updates on this?

**Comment by UnleashedMind** [comment]
@roychoo 
Thank you very much for your feedbacks. This is a bug. 
This PR does not completely resolve this bug, the copyBatch method need to be awaited. 
Could you please open an issue as bug, and let us fix it?

File: packages/amplify-frontend-javascript/lib/frontend-config-creator.js [hunk 1]
```diff
@@ -60,11 +60,11 @@ function createAWSExports(context, amplifyResources) {
       default: break;
     }
   });
-  generateAWSExportsFile(context, configOutput);
+  await generateAWSExportsFile(context, configOutput);
   return context;
 }
 
-function generateAWSExportsFile(context, configOutput) {
+async function generateAWSExportsFile(context, configOutput) {
```
Comment (roychoo) [inline packages/amplify-frontend-javascript/lib/frontend-config-creator.js]: just a note. this async may not be necessary, but it is more explicit that this is returning a promise.
Got a bit lost in the trail of function returning promises at the start

**Comment by roychoo** [comment]
Hi @UnleashedMind , yeah i had realised that. i had edit this PR with the related changes.
i also created an issue https://github.com/aws-amplify/amplify-cli/issues/1051 (edited the issue)

File: packages/amplify-frontend-javascript/lib/frontend-config-creator.js (hunk 1 above)
Comment (UnleashedMind) [reply packages/amplify-frontend-javascript/lib/frontend-config-creator.js]: you can put an `await' in front of the `amplify.copyBatch` in stead of `return`.

**Comment by UnleashedMind** [comment]
@roychoo 
`onCategoryOutputsChange` method is changed to an async function, make sure all the places that this function is invoked, there's an `await` keyword in front of it if we need to wait for its execution to complete. 
You may need to put additional `async` in front of other methods because of this, please check all the packages in the mono-repo and make sure that this is correctly set.

**Comment by roychoo** [comment]
ok will do

**Comment by roychoo** [comment]
@UnleashedMind i am done, do take a look and let me know how is it

**Comment by roychoo** [comment]
Thank you

**Pull request merged**